pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTAnyTypeDeclaration.java

```diff
@@ -6,33 +6,72 @@
 
 import java.util.List;
 
+
 /**
- * Groups enums, classes and interface declarations.
+ * Groups enum, class, annotation and interface declarations.
  *
  * @author Clément Fournier
  */
-public interface ASTAnyTypeDeclaration extends JavaQualifiableNode, AccessNode, JavaNode {
+public abstract class ASTAnyTypeDeclaration extends AbstractJavaAccessTypeNode implements JavaQualifiableNode, AccessNode, JavaNode {
```

Comment:
the change from interface to abstract class is also a breaking API change

Reply:
This seems to cause trouble. We could very much keep the abstract class and interface separate, which wouldn't be API breaking. We'd just insert an abstract class in the hierarchy.